src/win/fs.c

```diff
@@ -315,8 +328,137 @@ static int fs__readlink_handle(HANDLE handle,
     return -1;
   }
 
-  assert(target_ptr == NULL || *target_ptr == NULL);
-  return uv_utf16_to_wtf8(w_target, w_target_len, target_ptr, target_len_ptr);
+
+  if (w_target_wtf8_len_ptr)
+    *w_target_wtf8_len_ptr = uv_utf16_length_as_wtf8(w_target, w_target_len);
+
+  w_target_size = w_target_len + 1;
+  if (w_target_ptr) {
+    if (*w_target_ptr) { /* preallocated buffer */
+      assert(w_target_size_ptr);
+
+      /* return if buffer is too small */
+      if (*w_target_size_ptr < w_target_size) {
+        *w_target_size_ptr = w_target_size;
+        return UV_ENOBUFS;
+      }
+    } else { /* buffer needs to be allocated by us */
+     *w_target_ptr = uv__malloc(w_target_size * sizeof(WCHAR));
+
+     if (*w_target_ptr == NULL) {
+       if (w_target_size_ptr)
+         *w_target_size_ptr = w_target_size;
+
+       SetLastError(ERROR_OUTOFMEMORY);
+       return -1;
+     }
+    }
+
+    memcpy(*w_target_ptr, w_target, w_target_len * sizeof(WCHAR));
+    (*w_target_ptr)[w_target_len] = '\0';
+  }
+
+  if (w_target_size_ptr)
+    *w_target_size_ptr = w_target_len;
+
+
+  return 0;
+}
+
+/* CreateFile wrapper. This uses our own readlink logic if windows can't do it. */
+static HANDLE fs__create_file(WCHAR* path, DWORD desired_access, int do_lstat) {
+  HANDLE handle;
+  size_t target_buf_size = 512;
+  size_t target_path_size;
+  WCHAR initial_buf[512];
+  WCHAR* target_path = (WCHAR*)initial_buf;
+  int need_free = 0;
+  int err;
+
+  if (do_lstat) {
+    return CreateFileW(path,
+                       desired_access,
+                       FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                       NULL,
+                       OPEN_EXISTING,
+                       FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OPEN_REPARSE_POINT,
+                       NULL);
+  }
+
+  for (uint32_t i = 0; i < 32; i++) {
+
+    /* try to get a handle normally */
+    handle = CreateFileW(path,
+                         desired_access,
+                         FILE_SHARE_READ | FILE_SHARE_WRITE |
+                         FILE_SHARE_DELETE,
+                         NULL,
+                         OPEN_EXISTING,
+                         FILE_FLAG_BACKUP_SEMANTICS,
+                         NULL);
+
+    if (handle != INVALID_HANDLE_VALUE)
```

Comment:
Is there any useful error to check here to know if it was a failed reparse point operation? Seems like a python developer said it is ERROR_CANT_ACCESS_FILE, and should not be handled https://github.com/python/cpython/issues/104336#issuecomment-1540878585
```suggestion
    if (handle != INVALID_HANDLE_VALUE || GetLastError() != ERROR_CANT_ACCESS_FILE)
```

Although that also implies our mapping of it to EACCES was wrong, and it should instead be mapped to
>       ENOENT A directory component in pathname does not exist or is a dangling symbolic link.

Reply:
Yeah, even if it is inaccurate, we shouldn't change that errno mapping now